From aa5926bff922a331842f73b502d8f11a67b7cc3b Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Fri, 20 Oct 2017 12:03:54 +0300
Subject: [PATCH 134/767] staging: fsl-dpaa2/eth: No support for XDP header
 adjustment

We did guard against increases in header size, but we don't
support decapsulation either. Check that the data pointer
is not changed by the XDP program.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index 8038d986b5fa..2c351c1289c8 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -323,6 +323,9 @@ static u32 dpaa2_eth_run_xdp(struct dpaa2_eth_priv *priv,
 	xdp.data_hard_start = xdp.data;
 
 	xdp_act = bpf_prog_run_xdp(xdp_prog, &xdp);
+	if (xdp.data != xdp.data_hard_start)
+		dev_warn_once(dev, "XDP header adjustment is not supported");
+
 	switch (xdp_act) {
 	case XDP_PASS:
 		break;
-- 
2.17.0

